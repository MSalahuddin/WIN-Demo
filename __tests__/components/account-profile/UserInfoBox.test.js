import React from 'react';
import { render } from 'react-native-testing-library';

import MockProvider from '../../MockProvider';
import UserInfoBox from '../../../js/components/account-profile/UserInfoBox';

describe('Component UserInfoBox', () => {
  const vipBannerOnPress = jest.fn();

  const profileData = {
    userName: '',
    firstName: '',
    lastName: ''
  };

  test('renders correctly', () => {
    const output = render(
      <MockProvider>
        <UserInfoBox vipBannerOnPress={vipBannerOnPress} profileData={profileData} />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });

  test('renders correctly when isVip', () => {
    const output = render(
      <MockProvider value={{ isVip: true }}>
        <UserInfoBox vipBannerOnPress={vipBannerOnPress} profileData={profileData} />
      </MockProvider>
    );
    expect(output).toMatchSnapshot();
  });
});
