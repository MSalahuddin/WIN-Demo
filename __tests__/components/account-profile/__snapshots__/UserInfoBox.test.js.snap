// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component UserInfoBox renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
    ]
  }
>
  <Image
    source={null}
    style={
      Array [
        Object {
          "backgroundColor": "#BFBFBF",
          "borderColor": "#FFFFFF",
          "borderRadius": 88,
          "borderWidth": 1,
          "height": 176,
          "width": 176,
        },
      ]
    }
  />
  <Text
    alignCenter={false}
    allowFontScaling={true}
    color="#000000"
    fontFamily="SEMI BOLD"
    isUnderlined={false}
    isUppercase={false}
    marginTop={16}
    maxFontSizeMultiplier={1.3}
    size="XLARGE"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Calibre-Semibold",
          "fontSize": 36,
          "justifyContent": "center",
          "textAlign": "left",
          "textDecorationColor": "#000000",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
          "textTransform": "none",
        },
        Object {
          "color": "#222222",
          "marginTop": 32,
        },
      ]
    }
  >
    
  </Text>
  <Text
    alignCenter={false}
    allowFontScaling={true}
    color="#000000"
    fontFamily="BOLD"
    isUnderlined={false}
    isUppercase={false}
    maxFontSizeMultiplier={1.3}
    size="SMALL"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "CalibreBold",
          "fontSize": 27,
          "justifyContent": "center",
          "textAlign": "left",
          "textDecorationColor": "#000000",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
          "textTransform": "none",
        },
        Object {
          "color": "#222222",
        },
      ]
    }
  >
    Guest
  </Text>
</View>
`;

exports[`Component UserInfoBox renders correctly when isVip 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
    ]
  }
>
  <Image
    source={null}
    style={
      Array [
        Object {
          "backgroundColor": "#BFBFBF",
          "borderColor": "#FFFFFF",
          "borderRadius": 88,
          "borderWidth": 1,
          "height": 176,
          "width": 176,
        },
      ]
    }
  />
  <Text
    alignCenter={false}
    allowFontScaling={true}
    color="#000000"
    fontFamily="SEMI BOLD"
    isUnderlined={false}
    isUppercase={false}
    marginTop={16}
    maxFontSizeMultiplier={1.3}
    size="XLARGE"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Calibre-Semibold",
          "fontSize": 36,
          "justifyContent": "center",
          "textAlign": "left",
          "textDecorationColor": "#000000",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
          "textTransform": "none",
        },
        Object {
          "color": "#222222",
          "marginTop": 32,
        },
      ]
    }
  >
    
  </Text>
  <Text
    alignCenter={false}
    allowFontScaling={true}
    color="#000000"
    fontFamily="BOLD"
    isUnderlined={false}
    isUppercase={false}
    maxFontSizeMultiplier={1.3}
    size="SMALL"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "CalibreBold",
          "fontSize": 27,
          "justifyContent": "center",
          "textAlign": "left",
          "textDecorationColor": "#000000",
          "textDecorationLine": "none",
          "textDecorationStyle": "solid",
          "textTransform": "none",
        },
        Object {
          "color": "#222222",
        },
      ]
    }
  >
    Guest
  </Text>
</View>
`;
